src/commands/registerCommands.ts

```diff
@@ -139,49 +140,76 @@ for (const label of bindingTypes.keys()) {
-      canSelectMany: false,
-      openLabel: 'Select XSD or DTD file',
-      filters: {
-        'Grammar files': ['xsd', 'dtd']
+  };
+
+  const fileUri = await window.showOpenDialog(options);
+  if (fileUri && fileUri[0]) {
+    // The XSD, DTD has been selected, get the proper syntax for binding this grammar file in the XML document.
+    const identifier = TextDocumentIdentifier.create(documentURI.toString());
+    const grammarURI = fileUri[0];
+    try {
+      const result = await commands.executeCommand(ServerCommandConstants.ASSOCIATE_GRAMMAR_INSERT, identifier, grammarURI.toString(), bindingType);
+      // Insert the proper syntax for binding
+      const lspTextDocumentEdit = <TextDocumentEdit>result;
+      const workEdits = new WorkspaceEdit();
+      for (const edit of lspTextDocumentEdit.edits) {
+        workEdits.replace(documentURI, languageClient.protocol2CodeConverter.asRange(edit.range), edit.newText);
       }
+      workspace.applyEdit(workEdits); // apply the edits
+    } catch (error) {
+      window.showErrorMessage('Error during grammar binding: ' + error.message);
     };
+  }
+}
 
-    const fileUri = await window.showOpenDialog(options);
-    if (fileUri && fileUri[0]) {
-      // The XSD, DTD has been selected, get the proper syntax for binding this grammar file in the XML document.
-      const identifier = TextDocumentIdentifier.create(documentURI.toString());
-      const grammarURI = fileUri[0];
-      try {
-        const result = await commands.executeCommand(ServerCommandConstants.ASSOCIATE_GRAMMAR_INSERT, identifier, grammarURI.toString(), bindingType);
-        // Insert the proper syntax for binding
-        const lspTextDocumentEdit = <TextDocumentEdit>result;
-        const workEdits = new WorkspaceEdit();
-        for (const edit of lspTextDocumentEdit.edits) {
-          workEdits.replace(documentURI, languageClient.protocol2CodeConverter.asRange(edit.range), edit.newText);
-        }
-        workspace.applyEdit(workEdits); // apply the edits
-      } catch (error) {
-        window.showErrorMessage('Error during grammar binding: ' + error.message);
-      };
-    }
+/**
+ * Register commands used for associating grammar file (XSD,DTD) to a given XML file with CodeLens
+ *
+ * @param context the extension context
+ * @param languageClient the language server client
+ */
+function registerCodeLensAssociationCommands(context: ExtensionContext, languageClient: LanguageClient) {
+  context.subscriptions.push(commands.registerCommand(ClientCommandConstants.OPEN_BINDING_WIZARD, async (uriString: string) => {
+    grammarAssociationCommand(uriString, languageClient)
   }));
+}
 
-}
\ No newline at end of file
+/**
+ * Register commands used for associating grammar file (XSD,DTD) to a given XML file
+ *
+ * @param context the extension context
+ * @param languageClient the language server client
+ */
+ function registerAssociationCommands(context: ExtensionContext, languageClient: LanguageClient) {
```

Comment:
I think it would be a good idea to combine the registration of this and the code lens version into one function, since they are related.

Reply:
Will this be affected by one having the uri passed into the function? (i.e. the codelens one receives the uri properly, while my addition doesn't)

Reply:
Nope, its just combining the registration of the commands, not the commands themselves.